provider/azure/instance.go

```diff
@@ -283,7 +283,7 @@ func (inst *azureInstance) OpenPorts(ctx context.ProviderCallContext, machineId
 			nil, // abort channel
 		)
 		if err := <-errCh; err != nil {
-			return errors.Annotatef(err, "creating security rule for %s", securityRule)
+			return errors.Annotatef(err, "creating security rule for %+v", securityRule)
```

Comment:
Why would we want %+v here? What was wrong with %s?

Reply:
provider/azure/instance.go:286: Annotatef format %s has arg securityRule of wrong type github.com/Azure/azure-sdk-for-go/arm/network.SecurityRule

Looks like network.SecurityRule doesn't implement String()

Reply:
i ended up changing this one to %s and ruleName, matched better with the other log and error messages.